### PR TITLE
ENH: Add `level` parameter to compress_content_streams

### PR DESCRIPTION
**Pull request by MartinThoma**
Provide more options / details on how to reduce the file size with compression.

See #1910

## A small experiment

```
File    File name
size    The suffix is the compression level
-------------------
5321132 GeoTopo.pdf
9959402 out-0.pdf
5976025 out-1.pdf
5914204 out-2.pdf
5885818 out-3.pdf
5816263 out-4.pdf
5762359 out-5.pdf
5738259 out-6.pdf
5731877 out-7.pdf
5726121 out-8.pdf
5725267 out-9.pdf
```

Level 1 gives a very good improvement, but already level 2 might not be worth the CPU cycles.

Interestingly, the original is smaller than the best compression :warning:

### CODE REVIEW

**Review by pubpub-zz — APPROVED** [review]

File: docs/user/file-size.md [hunk 1]
```diff
@@ -96,6 +96,10 @@ with open("out.pdf", "wb") as f:
     writer.write(f)
 ```
 
+`page.compress_content_streams` uses [`zlib.compress`](https://docs.python.org/3/library/zlib.html#zlib.compress) and support the
```
Comment (stefan6419846) [inline docs/user/file-size.md:99]: "and supports the"

File: docs/user/file-size.md [hunk 2]
```diff
@@ -96,6 +96,10 @@ with open("out.pdf", "wb") as f:
...
+`level` paramter: `level=0` is no compression, `level=9` is the
```
Comment (stefan6419846) [inline docs/user/file-size.md:100]: "parameter"

Maybe rewording to

> `level=0` means no compression, `level=9` refers to the highest compression.

makes sense here.

**Comment by MartinThoma** [comment]
Thanks for the improvement suggestions @stefan6419846 and thank you for the review @pubpub-zz :pray:
